network/src/protocols/mod.rs

```diff
@@ -337,8 +338,8 @@ impl CKBProtocolContext for DefaultCKBProtocolContext {
             .filter_broadcast(target, self.proto_id, data)?;
         Ok(())
     }
-    fn disconnect(&self, peer_index: PeerIndex) -> Result<(), Error> {
-        self.p2p_control.disconnect(peer_index)?;
+    fn disconnect(&self, peer_index: PeerIndex, message: Option<&str>) -> Result<(), Error> {
```

Comment:
It seems unnecessary to wrap an `Option` around the message.

Reply:
Yes, use empty `""` directly. It is the same as `unwrap_or_default("")`.

Reply:
> Yes, use empty `""` directly. It is the same as `unwrap_or_default("")`.

OK for me. 
@jjyr 
What do you think?

Reply:
IMO we should always return a reason, but for now, it's OK to return an empty string.